File: php/sort/Bubble.php
<?php
require ("AbstractSort.php");
class Bubble extends AbstractSort
{
	public static function sort(array $input)
	{
		$length = count($input);
		for ($i = 0; $i < $length; $i++) {

			for ($j = 0; $j < $length - 1; $j++) {

				if ($input[$j] > $input[$j + 1]) {
					$tmp = $input[$j + 1];
					$input[$j + 1] = $input[$j];
					$input[$j] = $tmp;
				}
				
			}

		}

		return $input;
	}
}

Bubble::main();

File: python/sort/bubble.py
# -*- coding: utf-8 -*-
from sort import AbstractSort


class BubbleSort(AbstractSort):

    @classmethod
    def sort(cls, origin):
        length = len(origin)
        for i in range(0, length):
            j = 0
            while j < length - 1:
                if origin[j] > origin[j + 1]:
                        tmp = origin[j + 1]
                        origin[j + 1] = origin[j]
                        origin[j] = tmp
                j += 1

